src/main/java/org/spongepowered/common/mixin/core/entity/MixinEntity.java

```diff
@@ -251,6 +253,29 @@ public void firePostConstructEvents() {
         this.isConstructing = false;
     }
 
+    @Inject(method = "addPassenger(Lnet/minecraft/entity/Entity;)V", at = @At(value="HEAD"), cancellable = true)
+    public void onAddPassenger(net.minecraft.entity.Entity passenger, CallbackInfo ci) {
+        if (!this.worldObj.isRemote) {
+            MountEntityEvent event = SpongeEventFactory.createMountEntityEvent(Cause.of(NamedCause.source(passenger)), this);
+            SpongeImpl.postEvent(event);
+            Thread.dumpStack();
+            if(event.isCancelled()) {
+                ci.cancel();
+            }
+        }
+    }
+
+    @Inject(method = "removePassenger(Lnet/minecraft/entity/Entity;)V", at = @At(value="HEAD"))
+    public void onRemovePassenger(net.minecraft.entity.Entity passenger, CallbackInfo ci) {
+        if (!this.worldObj.isRemote) {
+            DismountEntityEvent event = SpongeEventFactory.createDismountEntityEvent(Cause.of(NamedCause.source(passenger)), this);
+            SpongeImpl.postEvent(event);
```

Comment:
I based this off of other events that I saw, and they separate it into two lines. I'll totally collapse mine into one line, though, since that makes sense.